### DISCUSSION TITLE
Creating directories that do not exist

### DISCUSSION THREAD

**Discussion by theskipster** [hairyhenderson/gomplate#1275] [Q&A] [answered]
Is there a way to force the creation of directories when supplying an "outputFiles:" section in gomplate.yaml?

If I use this myfile.gomplate.yaml file:

```
inputFiles:
  - ./legit/path/to/myfile.tmpl.yaml
outputFiles:
  - ./missing/dir/myfile.yaml
```

I get this error:

`ERR  error="failed to open output file './missing/dir/myfile.yaml' for writing: open ./missing/dir/myfile.yaml: no such file or directory"`

I would like to have it create the missing directory when it generates each of the outputFiles.

**Comment by hairyhenderson** [comment, marked as answer]
Hi @theskipster - Happy New Year! Sorry for the slow response, I've been on vacation 😉

If you use [`inputDir`](https://docs.gomplate.ca/config/#inputdir)/[`outputDir`](https://docs.gomplate.ca/config/#outputdir), gomplate will create the paths, but that isn't the behaviour for `outputFiles`. I'm not sure I can think of a good reason _not_ to automatically create the output directory, and it's a bit unexpected. I'll convert this to an issue and add this as a feature.

**Reply by theskipster** [reply]
Thanks, @hairyhenderson!  I don't need this much anymore, as I have moved on, but might still be useful.

**Comment by hairyhenderson** [comment]
I'll address this in #1277!
